### PR TITLE
feat: detect and validate inner queries in default_zero() functions

### PR DESCRIPTION
**Pull request by ianchesal**
- Add query parsing to detect default_zero() usage and extract inner queries
- Validate inner queries separately to catch metrics masked by default_zero()
- Support nested default_zero() calls with proper parentheses matching
- Fail linting when default_zero() masks invalid metrics or syntax errors
- Add comprehensive test suite with edge cases and nested scenarios
- Enhance logging with debug info for query analysis and validation

BREAKING CHANGE: Linter now fails for invalid metrics wrapped in default_zero()
that previously passed validation

Resolves issue where default_zero() masked invalid metrics in CI pipelines

### CODE REVIEW

File: ENHANCEMENT.md [hunk 1]
```diff
@@ -0,0 +1,67 @@
+# Enhanced default_zero() Validation
```
[the remark this answers is not in the document]
Comment (4ndypanda) [reply ENHANCEMENT.md:1]: Does it do anything with the agent/LLMs performance or is this for humans?

**Comment by ianchesal** [comment]
Process I used:

* First I told it to summarize the repo:
```
> summarize this repository for me
```

* And then I told it what I wanted it to do:
```
you are an expert in golang and datadog query writing. you can write clear datadog query yaml and exceptionally high quality golang software. we want to improve ths query linting that this golang code does on datadog metrics. specifically, when using `default_zero()` on an invalid metric, the linter will pass. we'd like the linter to fail if `default_zero()` is being called on an invalid metric. when you update the golang code to implement this lint check, add tests that prove that you've solved the problem as well.
```

It was on auto-pilot mode (I think they call it "smart approval") and did not stop to ask me for anything until it was done generating the code, the tests, running the tests, and writing a summary of what it had done.

I was using `goose` and `sonnet-4` for the LLM.

**Comment by ianchesal** [comment]
@kuzmik @4ndypanda the second commit to find multiple instances really blew up the complexity of the go code. I didn't read it very closely TBH. Would definitely not trust it without a good read from one of you.

The first commit to catch the simpler cases seems good to me though.

File: ENHANCEMENT.md (hunk 1 above)
[the remark this answers is not in the document]
Comment (ianchesal) [reply ENHANCEMENT.md:1]: i don't think it helps the LLM at all

**Comment by 4ndypanda** [comment]
> @kuzmik @4ndypanda the second commit to find multiple instances really blew up the complexity of the go code. I didn't read it very closely TBH. Would definitely not trust it without a good read from one of you.
> 
> The first commit to catch the simpler cases seems good to me though.

Looks like the agent is solving the problem by extracting all the individual metrics in the query, removing the `default_zero(...)` function wrapping them, and validating each one. But I can't tell from the code generated if it is exhaustive. E.g. maybe the query is invalid despite all the metrics being valid.

Seems like an easier implementation is just replacing all `default_zero(` with just `(` since having redundant parenthesis is fine in a query. So going from `(default_zero(default_zero(A))+default_zero(B))/default_zero(C)` to `( ( (A) ) + (B) ) / (C)` (I added spaces so it is easier to see).

**Review by kuzmik — APPROVED** [review]

**Comment by ianchesal** [comment]
@kuzmik @4ndypanda would you like me to have it do the `replace all default_zero() with ()` idea instead?

**Comment by ianchesal** [comment]
Going to try this with Claude instead.
